### PR TITLE
fixes history command behaviour

### LINKED ISSUE #55: History command prints and counts the last item twice
History command prints and counts the last item twice

flash a recent mbed-os-cliapp build.
type a command to console, for example help
type history:

History [3/32]:

### PR DESCRIPTION
**Pull request by jupe**
## Status
**READY**

Fixes #55

### CODE REVIEW

File: source/ns_cmdline.c [hunk 1]
```diff
@@ -1707,9 +1707,13 @@ int help_command(int argc, char *argv[])
 int history_command(int argc, char *argv[])
 {
     if (argc == 1) {
-        cmd_printf("History [%i/%i]:\r\n", (int)ns_list_count(&cmd.history_list), cmd.history_max_count);
+        int history_size = (int)ns_list_count(&cmd.history_list) - 1;
+        cmd_printf("History [%i/%i]:\r\n", history_size-1, cmd.history_max_count);
```
Comment (YifeiZuo01) [inline source/ns_cmdline.c]: 1. Max: `History [30/32]`
2. 
```
>help
>help
>help
>history:
History [1/32]:
[0]: help
[1]: history
```

Comment (jupe) [reply source/ns_cmdline.c]: :) not. Seems that it needs more tuning

Comment (jupe) [reply source/ns_cmdline.c]: in fact - it is expected. By default it does not store duplicates, see:
https://github.com/ARMmbed/mbed-client-cli/blob/master/source/ns_cmdline.c#L1351
maybe better to switch it off by default..

Comment (YifeiZuo01) [reply source/ns_cmdline.c]: actually, ubuntu terminal treats command history the same way  --> `bool noduplicate = true;`

**Comment by jupe** [comment]
@YifeiZuo01  please try again

**Comment by jupe** [comment]
@YifeiZuo01 now it behaviour same way than in linux - doesn't duplicate new entries..

**Pull request merged**